makeself.sh

```diff
@@ -581,7 +581,7 @@ fi
 tmparch="${TMPDIR:-/tmp}/mkself$$.tar"
 (
     cd "$archdir"
-    find . ! -type d | LC_ALL=C sort | xargs tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
+    find . ! -type d -print0 | LC_ALL=C sort | tr -d '\n' | xargs -0 tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
```

Comment:
This makes little sense to me. If you're using `-print0` with `find`, there'll be no `\n` in output hence `tr -d '\n'` is useless. What's worse, `sort` won't do anything meaningful with `\0`-separated list unless `-z` is used (not sure how standard it is but I can see it at least in my Gentoo and macOS boxes).

Reply:
These are good points.

Reply:
It doesn't appear that `sort -z` is very standard, however I wonder if `-t '\0'` would work well as an alternative.

Reply:
`-t '\0'` sets a *field* separator (not *record* separator) so doesn't help. Maybe this?

```suggestion
    find . ! -type d | LC_ALL=C sort | tr '\n' '\0' | xargs -0 tar $TAR_EXTRA -$TAR_ARGS "$tmparch"
```

Reply:
This might do it. Any reason for the `env`?

Reply:
I'm also wondering about the `env`. Does it isolate any better than a naked assignment to the shell variable?